zerver/views/webhooks/taiga.py

```diff
@@ -0,0 +1,231 @@
+
+
+        elif change_type == "is_blocked":
+            if message["change"]["diff"]["is_blocked"]["to"]:
+                event_type = "blocked"
+            else:
+                event_type = "unblocked"
+
+        elif change_type == "is_closed":
+            if message["change"]["diff"]["is_closed"]["to"]:
+                event_type = "closed"
+            else:
+                event_type = "reopened"
+
+        elif change_type == "user_story":
+            old, new = get_old_and_new_values(change_type, message)
+            event_type = "changed_us"
+            values.update({'old': old, 'new': new})
+
+        elif change_type in ["subject", 'name']:
+            event_type = 'renamed'
+            old, new = get_old_and_new_values(change_type, message)
+            values.update({'old': old, 'new': new})
+
+
+        elif change_type in ["estimated_finish", "estimated_start"]:
+            old, new = get_old_and_new_values(change_type, message)
+            if not old == new:
+                event_type = change_type
+                values.update({'old': old, 'new': new})
+            else:
+                # date hasn't changed
+                return None
+
+        elif change_type in ["priority", "severity", "type", "status"]:
+            event_type = 'changed_' + change_type
+            old, new = get_old_and_new_values(change_type, message)
+            values.update({'old': old, 'new': new})
+
+        else:
+            # we are not supporting this type of event
+            return None
+
+        evt.update({"type": message["type"], "event": event_type, "values": values})
+        return evt
+
+
+    def parse_message(message):
+        """ Parses the payload by delegating to specialized functions. """
+        events = []
+        if message["action"] in ['create', 'delete']:
+            events.append(parse_create_or_delete(message))
+        elif message["action"] == 'change':
+            if message["change"]["diff"]:
+                for value in message["change"]["diff"]:
+                    parsed_event = parse_change_event(value, message)
+                    if parsed_event: events.append(parsed_event)
+            if message["change"]["comment"]:
+                events.append(parse_comment(message))
+
+        return events
+
+    def generate_content(data):
+        """ Gets the template string and formats it with parsed data. """
+        try:
+            return templates[data['type']][data['event']] % data['values']
+        except KeyError:
+            return json_error("Unknown message")
+
+    try:
+        message = ujson.loads(request.body)
+    except ValueError:
+        return json_error("Malformed JSON input")
+
+    parsed_events = parse_message(message)
+
+    content = ""
+    for i in range(len(parsed_events)):
+        # there might be more than one event in payload
+        content += generate_content(parsed_events[i])
+        if i < len(parsed_events)-1: content += '\n'
```

Comment:
Is there a problem with having a trailing newline?  I think you might be able to just add a "\n" unconditionally.